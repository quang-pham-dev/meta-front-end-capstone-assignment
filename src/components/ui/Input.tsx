import * as React from "react";

export type InputProps = React.InputHTMLAttributes<HTMLInputElement>;

const Input = React.forwardRef<HTMLInputElement, InputProps>(
  ({ className, type, ...props }, ref) => {
    return (
      <input
        type={type}
        className={`${className || ""}`}
        ref={ref}
        {...props}
      />
    );
  }
);

Input.displayName = "Input";

export { Input };
