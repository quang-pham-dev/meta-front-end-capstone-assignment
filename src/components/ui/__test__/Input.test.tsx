import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Input } from "../Input";
import "@testing-library/jest-dom";

describe("Input", () => {
  it("renders with default props", () => {
    render(<Input />);
    const input = screen.getByRole("textbox");
    expect(input).toBeInTheDocument();
  });

  it("renders with custom type and className", () => {
    render(<Input type="number" className="custom-input" />);
    const input = screen.getByRole("spinbutton");
    expect(input).toHaveAttribute("type", "number");
    expect(input).toHaveClass("custom-input");
  });

  it("passes additional props to the input element", () => {
    render(<Input placeholder="Enter text" required />);
    const input = screen.getByPlaceholderText("Enter text");
    expect(input).toHaveAttribute("required");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <Input type="email" placeholder="Enter email" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
