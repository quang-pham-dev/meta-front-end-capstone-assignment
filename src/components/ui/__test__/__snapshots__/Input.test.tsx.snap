// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > matches snapshot 1`] = `
<DocumentFragment>
  <input
    class=""
    placeholder="Enter email"
    type="email"
  />
</DocumentFragment>
`;
